### PR TITLE
AssembleDownstreamDocumentation - print guide name

### PR DESCRIPTION
**Pull request by rsvoboda**
AssembleDownstreamDocumentation - print guide name

Motivation: we had 2 issues with docs during 2 last days, this helps to identify which guide has issues.

CC @sberyozkin

### CODE REVIEW

File: docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ public static void main(String[] args) throws Exception {
             Files.createDirectories(TARGET_ROOT_DIRECTORY);
 
             for (Path guide : guides) {
+                System.out.println("[INFO] Processing guide " + guide.getFileName());
```
Comment (sberyozkin) [inline docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:127]: Should `LOG.infof` be used instead ?

Comment (rsvoboda) [reply docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:127]: I had that in my initial version but the output is not nice, it is using JDK based logging which goes on 2 lines and is not aligned with the rest of output

```
[INFO] --- exec:3.0.0:exec (assemble-downstream-doc) @ quarkus-documentation ---
[INFO] Processing guide datasource.adoc
[INFO] Processing guide logging.adoc
[INFO] Processing guide security-architecture.adoc
[INFO] Processing guide security-authentication-mechanisms.adoc
[INFO] Processing guide security-authorize-web-endpoints-reference.adoc
[INFO] Processing guide security-basic-authentication-howto.adoc
[INFO] Processing guide security-basic-authentication-tutorial.adoc
[INFO] Processing guide security-basic-authentication.adoc
[INFO] Processing guide security-identity-providers.adoc
[INFO] Processing guide security-jpa.adoc
[INFO] Processing guide security-oidc-bearer-token-authentication-tutorial.adoc
[INFO] Processing guide security-oidc-bearer-token-authentication.adoc
[INFO] Processing guide security-oidc-code-flow-authentication-tutorial.adoc
[INFO] Processing guide security-oidc-code-flow-authentication.adoc
[INFO] Processing guide security-oidc-configuration-properties-reference.adoc
[INFO] Processing guide security-openid-connect-client-reference.adoc
[INFO] Processing guide security-overview.adoc
[INFO] Processing guide security-proactive-authentication.adoc
Oct 11, 2023 1:16:34 PM io.quarkus.docs.generation.AssembleDownstreamDocumentation main
INFO: Downstream documentation tree is available in: target/downstream-tree
Oct 11, 2023 1:16:34 PM io.quarkus.docs.generation.AssembleDownstreamDocumentation main
INFO: Downstream documentation listing is available in: target/downstream-files.txt
[INFO]
[INFO] --- asciidoctor:2.0.0:process-asciidoc (output-html) @ quarkus-documentation ---
2023-10-11T13:16:36.818+02:00 [main] WARN FilenoUtil : Native subprocess control requires open access to sun.nio.ch
Pass '--add-opens java.base/sun.nio.ch=org.jruby.dist' or '=org.jruby.core' to enable.
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 195 resources
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/scripting.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/kafka-dev-ui.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/command-mode-reference.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/grpc-xds.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/picocli.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-webauthn.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/deploying-to-google-cloud.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/tooling.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/dev-ui-v2.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/aws-lambda-snapstart.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/elasticsearch.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/logging.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/http-reference.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/writing-native-applications-tips.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/dev-services.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/building-my-first-extension.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/datasource.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/getting-started-testing.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/transaction.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/config-yaml.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/azure-functions.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/vertx-reference.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/smallrye-fault-tolerance.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/funqy-azure-functions-http.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/hibernate-orm-panache-kotlin.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-proactive-authentication.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/mutiny-primer.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/kafka.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/kubernetes-config.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/hibernate-reactive.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/qute.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/maven-tooling.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/pulsar-getting-started.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-jwt-build.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/deploying-to-azure-cloud.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/init-tasks.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/hibernate-search-orm-elasticsearch.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/mongodb.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/jreleaser.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/kafka-schema-registry-avro.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/spring-cache.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/build-analytics.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-identity-providers.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-openid-connect-client.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/jms.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/lra.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-vulnerability-detection.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/spring-boot-properties.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/spring-cloud-config-client.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/security-testing.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/blaze-persistence.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/capabilities.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/quarkus-reactive-architecture.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/opentelemetry.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/resteasy-reactive-migration.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/grpc-service-consumption.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/funqy-http.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/spring-scheduled.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/centralized-log-management.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/conditional-extension-dependencies.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/native-reference.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/aws-lambda-http.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/azure-functions-http.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/quarkus-maven-plugin.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/flyway.adoc
[INFO] Converted /Users/rsvoboda/tmp/quarkus/docs/target/asciidoc/sources/all-builditems.adoc
...
```

Comment (rsvoboda) [reply docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:127]: Thesame approach is used on many places in docs module
```
~/git/quarkus/docs    print.guide.name  git grep 'System.out.println("\[' 
src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:                System.out.println("[INFO] Processing guide " + guide.getFileName());
src/main/java/io/quarkus/docs/generation/CopyExampleSource.java:        System.out.println("[INFO] Output directory: " + copyExamples.outputPath);
src/main/java/io/quarkus/docs/generation/CopyExampleSource.java:        System.out.println("[INFO] Project root: " + copyExamples.rootPath);
src/main/java/io/quarkus/docs/generation/CopyExampleSource.java:                        System.out.println("[INFO] Reading: " + path);
src/main/java/io/quarkus/docs/generation/ReferenceIndexGenerator.java:        System.out.println("[INFO] Creating reference index generator: " + List.of(args));
src/main/java/io/quarkus/docs/generation/ReferenceIndexGenerator.java:        System.out.println("[INFO] Generating reference index");
src/main/java/io/quarkus/docs/generation/ReferenceIndexGenerator.java:        System.out.println("[INFO] Writing reference index file");
src/main/java/io/quarkus/docs/generation/ReferenceIndexGenerator.java:        System.out.println("[INFO] Transforming the source code");
src/main/java/io/quarkus/docs/generation/ReferenceIndexGenerator.java:        System.out.println("[INFO] Done");
src/main/java/io/quarkus/docs/generation/YamlMetadataGenerator.java:        System.out.println("[INFO] Creating YAML metadata generator: " + List.of(args));
src/main/java/io/quarkus/docs/generation/YamlMetadataGenerator.java:        System.out.println("[INFO] Generating metadata index");
src/main/java/io/quarkus/docs/generation/YamlMetadataGenerator.java:        System.out.println("[INFO] Writing metadata index and error files");
src/main/java/io/quarkus/docs/generation/YamlMetadataGenerator.java:        System.out.println("[INFO] Done");
```

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
As for the original problem, I barely translated to Java a bash script that was provided to me, as far as the regexps are concerned.

I have no idea what could make it stuck, tbh. Getting a thread dump might help to understand the problem.

**Comment by rsvoboda** [comment]
@gsmet here is [stuck-docs-thread-dump.txt](https://github.com/quarkusio/quarkus/files/12872005/stuck-docs-thread-dump.txt)

I dropped https://github.com/quarkusio/quarkus/pull/36384 changes and executed `mvn clean package -pl docs -DskipTests`. 
When the build got stuck I hit `Ctrl+\`.

**Comment by gsmet** [comment]
OK so my guess is that somehow the regexp is running forever (or is
extremely slow when the finishing part of the regexp is not around).

**Pull request merged**